### ISSUE TITLE
[Feature Request] A way to repeat the test

### ISSUE THREAD

**Issue by precondition** [lemnos/tt#2]
It would be nice to be able to repeat the exact same (sub-)test to work on accuracy, strive for a better WPM, etc.

I know that <kbd>Esc</kbd> "restarts the test" but in the case of random words, it generates a totally new set of words than what you were previously typing and in the case of fed-in text, it puts you back at the very first paragraph, regardless of what paragraph you were on when you pressed <kbd>Esc</kbd>.

In this regard, I like [wpm](https://github.com/cslarsen/wpm)'s navigation: space and arrows to move through (sub-)tests and esc to restart.

**Comment by lemnos** [comment]
> I know that Esc "restarts the test" but in the case of random words, it
generates a totally new set of words than what you were previously typing 

I think the default behaviour makes more sense and I'm not convinced this is useful enough to warrant a dedicated flag. For your own purposes you can achieve the desired result by generating the text yourself and relying on the default behaviour of custom text. E.G `shuf -n50 /usr/share/dict/words|tt`.

> and in the case of fed-in text, it puts you back at the very first paragraph,
regardless of what paragraph you were on when you pressed Esc.

I am considering implementing something like this, however I'm not sure how it should work. The current behaviour is to treat everything that gets piped in as a single test consisting of multiple paragraphs (which is why WPM is reported at the end). If the behaviour of escape was to change how would you modify the WPM accordingly?

> In this regard, I like wpm's navigation: space and arrows to move through (sub-)tests and esc to restart.

I am not familiar with it, but this sounds like it might be an interesting idea.

**Comment by lemnos** [comment]
Upon reflection I've realized what you propose is better default behaviour and implemented it. I have also added `-multi` which causes the program to treat treat each paragraph of the input as a separate test. It is worth noting that `tt` is flexible enough that it is possible to implement `-multi` externally

E.G

`fmt -w 1000 < input.txt|awk '!/^ *$/'|while read p; do echo $p|tt -oneshot || break; done`

however it is decidedly common enough to justify the inclusion of a dedicated flag.

If this addresses your use case feel free to close the issue or leave further comment :).
